### PR TITLE
[vscode] support CommentThread state property

### LINKED ISSUE #12231: [vscode] Support CommentThread state property
see https://code.visualstudio.com/updates/v1_75#_comment-thread-state and https://github.com/microsoft/vscode/issues/127473

**Summary:**
_This API controls whether a comment renders as resolved or unresolved and can affect filtering in the Comments view._

### PR DESCRIPTION
**Pull request by rschnekenbu**
#### What it does

_Add Comment#state optional property_
This adds the state optional property from Comment to improve vscode api coverage. It also displays the state in the Comment Thread widget

Fixes #12231 

Contributed on behalf of ST Microelectronics

#### How to test 
Note: there are currently some issues with menus, since theia 1.28.0. This is due to #11290, integrated in 1.28.0. Prior to this change, the comment sample works on theia 1.27.0. So to test this change, I cherry-picked my commit on top of 1.27.0 tag. 
This is the same case as for PR https://github.com/eclipse-theia/theia/pull/12007.
The issue https://github.com/eclipse-theia/theia/issues/12452 has been created to track those problems about menus.
1. Install forked extension [#comment-sample-0.0.1](https://github.com/rschnekenbu/vscode-extension-samples/releases/download/comment-with-state/forked-comment-api-example-0.0.1.vsix) vsix file - [Source](https://github.com/rschnekenbu/vscode-extension-samples/tree/rschnekenbu/comment-state/comment-sample)
2. Create a comment using the side bar while comparing 2 test files and reply with some comment with toggle or reply. 

3. The resolved state should be displayed along the thread label.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- [x] As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/main/browser/comments/comment-thread-widget.tsx [hunk 1]
```diff
@@ -231,7 +235,8 @@ export class CommentThreadWidget extends BaseWidget {
             if (this._commentThread.comments && this._commentThread.comments.length) {
                 const onlyUnique = (value: Comment, index: number, self: Comment[]) => self.indexOf(value) === index;
                 const participantsList = this._commentThread.comments.filter(onlyUnique).map(comment => `@${comment.userName}`).join(', ');
-                label = `Participants: ${participantsList}`;
+                const resolutionState = (this._commentThread.state === CommentThreadState.Resolved) ? '(Resolved)' : '(Unresolved)';
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/main/browser/comments/comment-thread-widget.tsx:238]: minor: the parentheses are unnecessary

```suggestion
                const resolutionState = this._commentThread.state === CommentThreadState.Resolved ? '(Resolved)' : '(Unresolved)';
```

**Review by vince-fugnitto — COMMENTED** [review]
The code looks good, I unfortunately cannot really test due to the menu issues you presented.
@rschnekenbu you might want to enable the ability for maintainers to perform edits so we can update the branch as needed so it satisfies the "out-of-date" check.

**Comment by rschnekenbu** [comment]
@vince-fugnitto , thanks a lot for your review!
 
Still would it make sense to accept the PR? We could add a check on the issue https://github.com/eclipse-theia/theia/issues/12452 for this API to be tested through the sample extension. The same should also be done for the already integrated Comment#timestamp  property support.

And also, unfortunately, I can't switch this ability to perform edit on my own, despite being the creator of the PR. I don't have access to the configuration of the PR. We already tried with Thomas for previous PRs, as he wanted to be able to rebase my PRs on latest main.

**Comment by rschnekenbu** [comment]
I rebased on top of master and I fixed the minor comment.

**Comment by vince-fugnitto** [comment]
@rschnekenbu I think that’s fine, we can revisit and retest this pull-request once the menu is fixed, did I understand that you wanted to fix it?

**Comment by rschnekenbu** [comment]
@vince-fugnitto, I won't have time to fix the menu issue before next release unfortunately. I already did some investigation, but the refactoring introduced on #11290 was too complex for me to understand the root cause of the issue.
As for the comment timestamp, I would be in favor of pushing the PR. Indeed, the feature is working as expected on 1.27 and the changes are isolated. Along with #12453 & #12456, that would allow next theia release to be API compatible with vscode 1.77.

**Review by vince-fugnitto — APPROVED** [review]
Let's go ahead and keep https://github.com/eclipse-theia/theia/issues/12452 in mind to properly review the changes once we fix the regression.

**Pull request merged**